File: src/main/java/br/com/report/service/impl/LogService.java
package br.com.report.service.impl;

import br.com.report.entity.Log;
import br.com.report.repository.*;
import br.com.report.service.interfaces.*;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class LogService implements LogServiceInterface {

    @Autowired
    private LogRepository logRepository;

    @Override
    public Log addLog(Log log) {
        return logRepository.save(log);
    }

    @Override
    public Optional<Log> findById(Long id) {
        return logRepository.findById(id);
    }

    @Override
    public void changeStatus(Log log) {
        logRepository.save(log);
    }

    @Override
    public List<Log> findAll() {
        return logRepository.findAll();
    }

    @Override
    public List<Log> findLogByEnvironment(String environment) {
        return logRepository.findLogByEnvironment(environment);
    }

    @Override
    public List<Log> findLogByEnvironmentAndOrderByLevel(String environment) {

        return logRepository.findLogByEnvironmentAndOrderByLevel(environment);
    }

    @Override
    public List<Log> findLogByEnvironmentAndOrderByEvent(String environment) {

        return logRepository.findLogByEnvironmentAndOrderByEvent(environment);
    }

    @Override
    public List<Log> findLogByEnvironmentAndSearchBy(String environment, String searchBy) {

        return logRepository.findLogByEnvironmentAndSearchBy(environment, searchBy);
    }

    @Override
    public List<Log> findLogByEnvironmentAndSearchByAndOrderByLevel(String environment, String searchBy){


        return logRepository.findLogByEnvironmentAndSearchByAndOrderByLevel(environment, searchBy);
    }

    @Override
    public List<Log> findLogByEnvironmentAndSearchByAndOrderByEvent(String environment, String searchBy){


        return logRepository.findLogByEnvironmentAndSearchByAndOrderByEvent(environment, searchBy);
    }

}

File: ErrorCenter-Angular/src/app/services/log.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { tap, catchError } from 'rxjs/operators';
import { Log } from '../model/log';
import { VariableEnvironmentService } from './variable-environment.service';

const httpOptions = {
  headers: new HttpHeaders({ 'Content-Type': 'application/json' })
};


@Injectable({
  providedIn: 'root'
})
export class LogService {

  private logUrl = this.variableEnvironment.url;
	//private logUrl = '/api';

  httpOptions = {
    headers: new HttpHeaders({ 'Content-Type': 'application/json' })
  };

  header = {
    headers: new HttpHeaders()
      .set('Authorization',  sessionStorage.getItem('token'))
  }

  constructor(private http: HttpClient, private variableEnvironment : VariableEnvironmentService) { }

  /**GET logs */
  getLogs(){
    console.log("GET DOS LOGS");
    return this.http.get(this.logUrl + "/api/logs", this.header);
  }

  // addLog(Log){
  //   return this.http.post(this.logUrl + "logs", Log);
  // }

  /**Get log by Id */
  getLogById(id){
    return this.http.get(this.logUrl + "/api/logs/id/" + id, this.header);
  }

  changeStatus(id, body) {
    return this.http.put(this.logUrl + "/api/logs/status/"+ id, body, this.header);
  }

  findLogByEnvironment(environment: String){
    return this.http.get(this.logUrl + "/api/logs/" + environment, this.header);
  }

  findLogByEnvironmentAndOrderByLevel(environment: String, orderBy: String){
    return this.http.get(this.logUrl + "/api/logs/envOrdLev/" + environment, this.header);
  }

  findLogByEnvironmentAndOrderByEvent(environment: String, orderBy: String){
    return this.http.get(this.logUrl + "/api/logs/envOrdEve/" + environment, this.header);
  }

  findLogByEnvironmentAndSearchByAndOrderByLevel(environment: String, searchBy: String){
    return this.http.get(this.logUrl + "/api/logs/envOrdLev/" + environment + "/" + searchBy, this.header);
  }

  findLogByEnvironmentAndSearchByAndOrderByEvent(environment: String, searchBy: String){
    return this.http.get(this.logUrl + "/api/logs/envOrdEve/" + environment + "/" + searchBy, this.header);
  }

  findLogByEnvironmentAndSearchBy(environment: String, searchBy: String){
    return this.http.get(this.logUrl + "/api/logs/" + environment + "/" + searchBy, this.header);
  }
  
}
